dmarcts-report-viewer.php

```diff
@@ -169,14 +169,15 @@ function tmpl_reportData($reportnumber, $allowed_reports, $host_lookup = 1) {
 	return implode("\n  ",$reportdata);
 }
 
-function tmpl_page ($body, $reportid, $host_lookup = 1, $dom_select, $domains = array() ) {
+function tmpl_page ($body, $reportid, $host_lookup = 1, $sort_order, $dom_select, $domains = array() ) {
 	$html       = array();
 	$url_hswitch = ( $reportid ? "?report=$reportid&hostlookup=" : "?hostlookup=" )
-                . ($host_lookup ? "0" : "1" )
+		. ($host_lookup ? "0" : "1" )
+                . ( "&sortorder=" ) . ($sort_order)
                 . (isset($dom_select) && $dom_select <> "" ? "&d=$dom_select" : "" )
                 ;
-	$url_dswitch = "?hostlookup=" . ($host_lookup ? "1" : "0" ); // drop selected report on domain switch
-
+        $url_dswitch = "?hostlookup=" . ($host_lookup ? "1" : "0" ) . "&sortorder=" . ($sort_order); // drop selected report on domain switch
+        $url_sswitch = "?hostlookup=" . ($host_lookup) . "&sortorder=" . ($sort_order ? "0" : "1" );
```

Comment:
Here it is needed tp maintain the currently selected domain or a selected domain is lost. 
`        $url_sswitch = "?hostlookup=" . ($host_lookup) . "&sortorder=" . ($sort_order ? "0" : "1" ) . (isset($dom_select) && $dom_select <> "" ? "&d=$dom_select" : "" );`


This simple switching design works only nice with few variables someday we should do a recode with a function to generate the new url dynamicly.

Reply:
also the README.md needs an update for the new config variable.